token/program-2022/src/extension/scaled_ui_amount/mod.rs

```diff
@@ -0,0 +1,341 @@
+#[cfg(feature = "serde-traits")]
+use serde::{Deserialize, Serialize};
+use {
+    crate::extension::{Extension, ExtensionType},
+    bytemuck::{Pod, Zeroable},
+    solana_program::program_error::ProgramError,
+    spl_pod::{optional_keys::OptionalNonZeroPubkey, primitives::PodI64},
+};
+
+/// Scaled UI amount extension instructions
+pub mod instruction;
+
+/// Scaled UI amount extension processor
+pub mod processor;
+
+/// `UnixTimestamp` expressed with an alignment-independent type
+pub type UnixTimestamp = PodI64;
+
+/// `f64` type that can be used in `Pod`s
+#[cfg_attr(feature = "serde-traits", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde-traits", serde(from = "f64", into = "f64"))]
+#[derive(Clone, Copy, Debug, Default, PartialEq, Pod, Zeroable)]
+#[repr(transparent)]
+pub struct PodF64(pub [u8; 8]);
```

Comment:
Was there a reason we define this type here rather than in `spl_pod::primitives` like the other primitive pod types?

Reply:
Yeah, two reasons:

1. We should discourage floats in on-chain programs, so if we put it in a general library people might feel compelled to use it
2. Whenever there's a change to spl-pod, it's very annoying to publish all the new crates

Reply:
Okay, makes sense to me!